client/dave_client_test.go

```diff
@@ -0,0 +1,84 @@
+package client
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/Gamebuildr/Dave/pkg/scaler"
+)
+
+type MockWatcher struct {
+	messagecount int
+}
+
+func (watcher MockWatcher) ReadQueueMessages(url string) (int, error) {
```

Comment:
I think this should be ReadQueueMessagesCount or similar, it's slightly misleading and not obvious that it's going to return the count/